src/dialect/mod.rs

```diff
@@ -300,13 +304,172 @@ pub trait Dialect: Debug + Any {
+            Token::Word(w) if w.keyword == Keyword::OPERATOR => Ok(BETWEEN_PREC),
+            Token::Word(w) if w.keyword == Keyword::DIV => Ok(MUL_DIV_MOD_OP_PREC),
+            Token::Eq
+            | Token::Lt
+            | Token::LtEq
+            | Token::Neq
+            | Token::Gt
+            | Token::GtEq
+            | Token::DoubleEq
+            | Token::Tilde
+            | Token::TildeAsterisk
+            | Token::ExclamationMarkTilde
+            | Token::ExclamationMarkTildeAsterisk
+            | Token::DoubleTilde
+            | Token::DoubleTildeAsterisk
+            | Token::ExclamationMarkDoubleTilde
+            | Token::ExclamationMarkDoubleTildeAsterisk
+            | Token::Spaceship => Ok(EQ_PREC),
+            Token::Pipe => Ok(PIPE_PREC),
+            Token::Caret | Token::Sharp | Token::ShiftRight | Token::ShiftLeft => Ok(CARET_PREC),
+            Token::Ampersand => Ok(AMPERSAND_PREC),
+            Token::Plus | Token::Minus => Ok(PLUS_MINUS_PREC),
+            Token::Mul | Token::Div | Token::DuckIntDiv | Token::Mod | Token::StringConcat => {
+                Ok(MUL_DIV_MOD_OP_PREC)
+            }
+            Token::DoubleColon
+            | Token::ExclamationMark
+            | Token::LBracket
+            | Token::Overlap
+            | Token::CaretAt => Ok(DOUBLE_COLON_PREC),
+            // Token::Colon if (self as dyn Dialect).is::<SnowflakeDialect>() => Ok(DOUBLE_COLON_PREC),
+            Token::Arrow
+            | Token::LongArrow
+            | Token::HashArrow
+            | Token::HashLongArrow
+            | Token::AtArrow
+            | Token::ArrowAt
+            | Token::HashMinus
+            | Token::AtQuestion
+            | Token::AtAt
+            | Token::Question
+            | Token::QuestionAnd
+            | Token::QuestionPipe
+            | Token::CustomBinaryOperator(_) => Ok(PG_OTHER_PREC),
+            _ => Ok(UNKNOWN_PREC),
+        }
+    }
+
     /// Dialect-specific statement parser override
     fn parse_statement(&self, _parser: &mut Parser) -> Option<Result<Statement, ParserError>> {
         // return None to fall back to the default behavior
         None
     }
+
+    /// The following precedence values are used directly by `Parse` or in dialects,
+    /// so have to be made public by the dialect.
+    fn prec_double_colon(&self) -> u8 {
+        DOUBLE_COLON_PREC
+    }
+
+    fn prec_mul_div_mod_op(&self) -> u8 {
+        MUL_DIV_MOD_OP_PREC
+    }
+
+    fn prec_plus_minus(&self) -> u8 {
+        PLUS_MINUS_PREC
+    }
+
+    fn prec_between(&self) -> u8 {
+        BETWEEN_PREC
+    }
+
+    fn prec_like(&self) -> u8 {
+        LIKE_PREC
+    }
+
+    fn prec_unary_not(&self) -> u8 {
+        UNARY_NOT_PREC
+    }
+
+    fn prec_unknown(&self) -> u8 {
+        UNKNOWN_PREC
+    }
 }
 
+// Define the lexical Precedence of operators.
+//
+// Uses (APPROXIMATELY) <https://www.postgresql.org/docs/7.0/operators.htm#AEN2026> as a reference
```

Comment:
This statement really isn't true, hence I added `APPROXIMATELY`.

We could rewrite to "was originally inspired by" or something?

Reply:
I think this is fine